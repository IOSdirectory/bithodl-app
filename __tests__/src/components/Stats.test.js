import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Stats from '../../../src/components/Stats';

jest.mock('../../../src/containers/BtcLabelContainer', () => 'BtcLabelContainer');

describe('Stats', () => {
  let fakeItems;

  beforeEach(() => {
    fakeItems = {
      '4be67eb7-6715-4d0c-bd8f-a2e85eaa2075': {
        id: '4be67eb7-6715-4d0c-bd8f-a2e85eaa2075',
        hash: '1JNHNJftiWu57UVpp2cjrNNHCeLwbNy3Zt',
        script: 'OP_RETURN',
        amount: 2000000,
        createdAt: 1471651200,
        unlockAt: 1482192000
      },
      'abbefad4-9faf-4f68-9987-9e88caaa866f': {
        id: 'abbefad4-9faf-4f68-9987-9e88caaa866f',
        hash: '1JNHNJftiWu57UVpp2cjrNNHCeLwbNy3Zt',
        script: 'OP_RETURN',
        amount: 1000000,
        createdAt: 1471351200,
        unlockAt: 1482152000
      },
      'e9474dd7-6ad3-4df1-afb2-1de3c477cc55': {
        id: 'e9474dd7-6ad3-4df1-afb2-1de3c477cc55',
        hash: '1JNHNJftiWu57UVpp2cjrNNHCeLwbNy3Zt',
        script: 'OP_RETURN',
        amount: 3000000,
        createdAt: 1471371221,
        unlockAt: 1482352139
      },
      'bf7fb80e-f138-4bba-8d62-fcec289ca2f8': {
        id: 'bf7fb80e-f138-4bba-8d62-fcec289ca2f8',
        hash: '1JNHNJftiWu57UVpp2cjrNNHCeLwbNy3Zt',
        script: 'OP_RETURN',
        amount: 4000000,
        createdAt: 1471371221,
        unlockAt: 1482352139
      }
    };
  });

  it('renders correctly with items', () => {
    const tree = renderer.create(
      <Stats items={fakeItems} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly without items', () => {
    const tree = renderer.create(
      <Stats />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
