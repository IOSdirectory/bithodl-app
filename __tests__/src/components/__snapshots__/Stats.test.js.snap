// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats renders correctly with items 1`] = `
<View
  style={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    numberOfLines={undefined}
    style={
      Array [
        Object {
          "color": "#504F63",
          "fontFamily": "Avenir Next",
          "fontWeight": "400",
        },
        Object {
          "color": "#D49146",
          "fontSize": 11,
          "fontWeight": "600",
          "letterSpacing": 1,
        },
      ]
    }
  >
    HODLING
  </Text>
  <BtcLabelContainer
    amount={10000000}
    style={
      Object {
        "color": "#504F63",
        "fontSize": 34,
        "letterSpacing": -0.82,
      }
    }
  />
</View>
`;

exports[`Stats renders correctly without items 1`] = `
<View
  style={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    numberOfLines={undefined}
    style={
      Array [
        Object {
          "color": "#504F63",
          "fontFamily": "Avenir Next",
          "fontWeight": "400",
        },
        Object {
          "color": "#D49146",
          "fontSize": 11,
          "fontWeight": "600",
          "letterSpacing": 1,
        },
      ]
    }
  >
    HODLING
  </Text>
  <BtcLabelContainer
    amount={0}
    style={
      Object {
        "color": "#504F63",
        "fontSize": 34,
        "letterSpacing": -0.82,
      }
    }
  />
</View>
`;
